phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/init/RestoreWalletView.kt

```diff
@@ -41,88 +49,229 @@ import fr.acinq.phoenix.controllers.init.RestoreWallet
+                                Text(stringResource(id = R.string.restore_in_progress))
                             }
                         }
-                        is RestoreWallet.Model.InvalidMnemonics -> {
-                            Text(stringResource(R.string.restore_error))
+
+                        LazyRow(
+                            horizontalArrangement = Arrangement.spacedBy(8.dp)
+                        ) {
+
+                            // Add 5 items
+                            items(filteredWords) {
+
+                                val apiString = AnnotatedString.Builder()
+                                apiString.pushStyle(
+                                    style = SpanStyle(
+                                        textDecoration = TextDecoration.Underline
+                                    )
+                                )
+                                apiString.append(it)
+                                Text(
+                                    modifier = Modifier.clickable(enabled = true) {
+                                        if (selectedWords.count() < 12) {
+                                            vm.addWord(it)
+                                            wordsInput = ""
+                                            postIntent(RestoreWallet.Intent.FilterWordList(predicate = wordsInput))
+                                        }
+                                    },
+                                    text = apiString.toAnnotatedString(),
+
+                                    )
+                            }
                         }
-                        is RestoreWallet.Model.ValidMnemonics -> {
-                            val writingState = vm.writingState
-                            if (writingState is WritingSeedState.Error) {
-                                Text(stringResource(id = R.string.autocreate_error, writingState.e.localizedMessage ?: writingState.e::class.java.simpleName))
-                            } else {
-                                Text(stringResource(R.string.restore_in_progress))
-                                LaunchedEffect(keyState) {
-                                    vm.writeSeed(context, wordsInput.split(" "), false, onSeedWritten)
+
+                        Column() {
+
+                            Spacer(Modifier.height(24.dp))
+                            Column(
+                                modifier = Modifier.fillMaxWidth(),
+                                horizontalAlignment = Alignment.CenterHorizontally
+                            ) {
+                                HSeparator()
+                            }
+                            Spacer(Modifier.height(24.dp))
+
+                            Row {
+                                Column(Modifier.weight(1f)) {
+
+                                    for (index in 0..5) {
+                                        Row(verticalAlignment = Alignment.CenterVertically) {
```

Comment:
This Row should be made into a composable, if only for improved readability, but also because this code is duplicated below.